sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/implementation/TablesJacksonSerializer.java

```diff
@@ -27,71 +28,107 @@
+                } else {
+                    // This is a single-entity response where the user just happened to use the key "value".
+                    if (singleValue == null) {
+                        singleValue = new HashMap<>();
+                    }
+                    singleValue.put(fieldName, getEntityFieldAsObject(node, fieldName));
                 }
-
-                throw logger.logExceptionAsError(
-                    new UnsupportedOperationException("Multiple return values not supported yet."));
+            } else {
+                // This is a single-entity response.
+                if (singleValue == null) {
+                    singleValue = new HashMap<>();
+                }
+                singleValue.put(fieldName, getEntityFieldAsObject(node, fieldName));
             }
-
-            rootObject.put(entry.getKey(), entry.getValue().asText());
         }
 
-        values.add(0, rootObject);
+        if (singleValue != null) {
+            if (values.size() > 0) {
+                throw logger.logExceptionAsError(new IllegalStateException(
+                    "Unexpected response format. Response containing a 'value' array must not contain other properties."
+                ));
+            }
+            values.add(singleValue);
+        }
 
         return (U) new TableEntityQueryResponse()
             .setOdataMetadata(odataMetadata)
             .setValue(values);
     }
 
-    @Override
-    public <U> U deserialize(HttpHeaders httpHeaders, Type type) throws IOException {
-        return super.deserialize(httpHeaders, type);
+    private Map<String, Object> getEntityFieldsAsMap(JsonNode node) {
+        Map<String, Object> result = new HashMap<>();
+        for (Iterator<String> it = node.fieldNames(); it.hasNext(); ) {
+            String fieldName = it.next();
+            result.put(fieldName, getEntityFieldAsObject(node, fieldName));
+        }
+        return result;
+    }
+
+    private Object getEntityFieldAsObject(JsonNode parentNode, String fieldName) {
+        JsonNode valueNode = parentNode.get(fieldName);
+        if (!TablesConstants.METADATA_KEYS.contains(fieldName) && !fieldName.endsWith("@odata.type")) {
+            JsonNode typeNode = parentNode.get(fieldName + "@odata.type");
+            if (typeNode != null) {
+                String type = typeNode.asText();
+                switch (type) {
+                    case "Edm.DateTime":
+                        try {
+                            return OffsetDateTime.parse(valueNode.asText());
+                        } catch (DateTimeParseException e) {
+                            throw logger.logExceptionAsError(new IllegalArgumentException(String.format(
+                                "'%s' value is not a valid OffsetDateTime.", TablesConstants.TIMESTAMP_KEY), e));
+                        }
+                    default:
+                        logger.warning("'%s' value has unknown OData type '%s'", fieldName, type);
```

Comment:
The placeholder for these is `{}` rather than `%s`

Reply:
Fixed.